src/pkg/services/m365/custom/drive_item.go

```diff
@@ -0,0 +1,339 @@
+func (c *driveItem) SetParentReference(parent itemReferenceable) {
+	c.parentRef = parent
+}
+
+func (c *driveItem) GetCreatedBy() identitySetable {
+	return c.createdBy
+}
+
+func (c *driveItem) GetCreatedByUser() userable {
+	return c.createdByUser
+}
+
+func (c *driveItem) GetLastModifiedByUser() userable {
+	return c.lastModifiedByUser
+}
+
+func (c *driveItem) GetCreatedDateTime() *time.Time {
+	return &c.createdDateTime
+}
+
+func (c *driveItem) GetLastModifiedDateTime() *time.Time {
+	return &c.lastModifiedDateTime
+}
+
+func (c *driveItem) GetAdditionalData() map[string]interface{} {
+	return c.additionalData
+}
+
+type (
+	malwareable interface {
+		GetDescription() *string
+	}
+	fileItemable interface {
+		GetMimeType() *string
+	}
+	itemReferenceable interface {
+		GetPath() *string
+		GetId() *string
+		GetName() *string
+		GetDriveId() *string
+	}
+	identitySetable interface {
+		GetUser() identityable
+	}
+	identityable interface {
+		GetAdditionalData() map[string]interface{}
+	}
+	userable interface {
+		GetId() *string
+	}
+)
+
+// Concrete implementations
+
+var _ malwareable = &malware{}
+
+type malware struct {
+	description string
+}
+
+func (m *malware) GetDescription() *string {
+	return &m.description
+}
+
+var _ fileItemable = &fileItem{}
+
+type fileItem struct {
+	mimeType string
+}
+
+func (f *fileItem) GetMimeType() *string {
+	return &f.mimeType
+}
+
+var _ itemReferenceable = &parentRef{}
+
+type parentRef struct {
+	path    string
+	id      string
+	name    string
+	driveID string
+}
+
+func (pr *parentRef) GetPath() *string {
+	return &pr.path
+}
+
+func (pr *parentRef) GetId() *string {
+	return &pr.id
+}
+
+func (pr *parentRef) GetName() *string {
+	return &pr.name
+}
+
+func (pr *parentRef) GetDriveId() *string {
+	return &pr.driveID
+}
+
+var _ identitySetable = &identitySet{}
+
+type identitySet struct {
+	identity identityable
+}
+
+func (iis *identitySet) GetUser() identityable {
+	return iis.identity
+}
+
+var _ identityable = &identity{}
+
+type identity struct {
+	additionalData map[string]interface{}
+}
+
+func (iu *identity) GetAdditionalData() map[string]interface{} {
+	return iu.additionalData
+}
+
+var _ userable = &user{}
+
+type user struct {
+	id string
+}
+
+func (u *user) GetId() *string {
+	return &u.id
+}
+
+// TODO(pandeyabs): This is duplicated from collection/drive package.
+// Move to common pkg
+var downloadURLKeys = []string{
+	"@microsoft.graph.downloadUrl",
+	"@content.downloadUrl",
+}
+
+func ToLiteDriveItemable(item models.DriveItemable) LiteDriveItemable {
+	if item == nil {
+		return nil
+	}
+
+	di := &driveItem{
+		id:                   strings.Clone(ptr.Val(item.GetId())),
+		name:                 strings.Clone(ptr.Val(item.GetName())),
+		size:                 ptr.Val(item.GetSize()),
+		createdDateTime:      ptr.Val(item.GetCreatedDateTime()),
+		lastModifiedDateTime: ptr.Val(item.GetLastModifiedDateTime()),
+	}
+
+	if item.GetFolder() != nil {
+		di.folder = &struct{}{}
```

Comment:
Would it be useful to have a func in `common/ptr` for this?  Something like:
```suggestion
	di.folder = ptr.EmptyOrNil(item.GetFolder())
```

Reply:
`&struct{}{}`.  I don't really have a better word for it.  `StructOrNil`?  Doesn't really matter all that much, mostly wondering if you think this would be valuable to centralize outside of this file.

Reply:
I don't see a huge value add tbh. I'm going to skip this.